src/components/Menu/index.tsx

```diff
@@ -163,6 +181,39 @@ const ToggleMenuItem = styled.button`
 
 const CODE_LINK = 'https://github.com/Uniswap/uniswap-interface'
 
+function LanguageMenu({ close }: { close: () => void }) {
+  const activeLocale = useActiveLocale()
+  const location = useLocation()
+  const qs = useParsedQueryString()
+
+  return (
+    <MenuFlyout>
+      <ToggleMenuItem onClick={close}>
+        <ChevronLeft size={16} />
+      </ToggleMenuItem>
+      {SUPPORTED_LOCALES.map((locale) => (
+        <InternalLinkMenuItem
```

Comment:
you could put this into a shared component like `SwitchLocaleLink<{ locale: SupportedLocale }>`

Reply:
or write a hook `useLocationLinkProps(locale: SupportedLocale): { to: Location, onClick: () => void }`

Reply:
refactored. Did I overcomplicate the hooks?